### ISSUE TITLE
Unique breaks duplicate button

### ISSUE THREAD

**Issue by TomSmedley** [payloadcms/payload#1243]
# Bug Report

https://www.loom.com/share/5808859ff5ed4f0a86b0fe3b868c44f9

## Current Behavior

If you create a unique field, in this case slug, it will prevent Duplicate from working. Removing this field as seen in the video above will still cause this issue to occur, also removing `unique: true` from the schema config causes this issue to occur.

## Expected Behavior

It either should append the unique field with it's first unique value such as `slug-name-1` or `slug-name-2` if 1 already exists. Or it should leave the field empty.

## Possible Solution

Create a unique index for unique fields allowing the duplicate functionality to not break.

## Steps to Reproduce

1. Create a collection with a unique field
2. Try to duplicate
3. Remove the field
4. Try to duplicate

**Comment by zhounewz** [comment]
@jmikrut  In order to solve this problem, I upgraded the payload from 1.1.7 to 1.1.11, but I encountered the following error：
![image](https://user-images.githubusercontent.com/16400882/195781811-869427bf-02d3-4c0e-8b67-09c8d0066a07.png)

**Comment by zhounewz** [comment]
@jmikrut  How to determine if `${data.uniqueText}-copy` is a duplicate, because it is possible

<img width="1075" alt="image" src="https://user-images.githubusercontent.com/16400882/196839358-0b90861a-02a3-4f0d-b237-a61b1b37361e.png">

**Comment by TomSmedley** [comment]
> @jmikrut How to determine if `${data.uniqueText}-copy` is a duplicate, because it is possible
> 
> <img alt="image" width="1075" src="https://user-images.githubusercontent.com/16400882/196839358-0b90861a-02a3-4f0d-b237-a61b1b37361e.png">

It's not possible because it will always append with `-copy` so it will become `-copy-copy`.

**Comment by zhounewz** [comment]
> > @jmikrut How to determine if `${data.uniqueText}-copy` is a duplicate, because it is possible
> > <img alt="image" width="1075" src="https://user-images.githubusercontent.com/16400882/196839358-0b90861a-02a3-4f0d-b237-a61b1b37361e.png">
> 
> It's not possible because it will always append with `-copy` so it will become `-copy-copy`.

Let me show you a specific scenario:
Assuming that there is an article collection, there is a uniqueText field that specifies unique as true;
Step 1: Perform a duplicate operation on the original article. At this time, the uniqueText field of the newly generated article becomes -copy.
Step 2: Go back to the original article again, pay attention! Here is to return to the original article and continue to perform the duplicate operation. An error will be reported at this time. Because for uniqueText fields, -copy already exists.

**Comment by TomSmedley** [comment]
> > > @jmikrut How to determine if `${data.uniqueText}-copy` is a duplicate, because it is possible
> > > <img alt="image" width="1075" src="https://user-images.githubusercontent.com/16400882/196839358-0b90861a-02a3-4f0d-b237-a61b1b37361e.png">
> > 
> > 
> > It's not possible because it will always append with `-copy` so it will become `-copy-copy`.
> 
> Let me show you a specific scenario: Assuming that there is an article collection, there is a uniqueText field that specifies unique as true; Step 1: Perform a duplicate operation on the original article. At this time, the uniqueText field of the newly generated article becomes `-copy`. Step 2: **Go back to the original article again, pay attention! Here is to return to the original article** and continue to perform the duplicate operation. An error will be reported at this time. Because for uniqueText fields, `-copy` already exists.

Good point, you would use the Local API to check whether that slug already exists before returning the data. https://payloadcms.com/docs/local-api/overview#accessing-payload

**Comment by zhounewz** [comment]
> > > > @jmikrut How to determine if `${data.uniqueText}-copy` is a duplicate, because it is possible
> > > > <img alt="image" width="1075" src="https://user-images.githubusercontent.com/16400882/196839358-0b90861a-02a3-4f0d-b237-a61b1b37361e.png">
> > > 
> > > 
> > > It's not possible because it will always append with `-copy` so it will become `-copy-copy`.
> > 
> > 
> > Let me show you a specific scenario: Assuming that there is an article collection, there is a uniqueText field that specifies unique as true; Step 1: Perform a duplicate operation on the original article. At this time, the uniqueText field of the newly generated article becomes `-copy`. Step 2: **Go back to the original article again, pay attention! Here is to return to the original article** and continue to perform the duplicate operation. An error will be reported at this time. Because for uniqueText fields, `-copy` already exists.
> 
> Good point, you would use the Local API to check whether that slug already exists before returning the data. https://payloadcms.com/docs/local-api/overview#accessing-payload

One problem is that the parameters injected in the beforeDuplicate hook are only the data and locale fields, and the payload cannot be obtained for local api calls. In contrast, in the beforeValidate hook, I can get the payload (`req.payload.find(....)`) through the injected req parameter and make a local api call.

![image](https://user-images.githubusercontent.com/16400882/196977332-b97f28b1-e4eb-4729-a929-baa367abf7e5.png)
![image](https://user-images.githubusercontent.com/16400882/196978249-3a785708-279a-4628-8d93-697383c2b562.png)

**Comment by TomSmedley** [comment]
> > > > > @jmikrut How to determine if `${data.uniqueText}-copy` is a duplicate, because it is possible
> > > > > <img alt="image" width="1075" src="https://user-images.githubusercontent.com/16400882/196839358-0b90861a-02a3-4f0d-b237-a61b1b37361e.png">
> > > > 
> > > > 
> > > > It's not possible because it will always append with `-copy` so it will become `-copy-copy`.
> > > 
> > > 
> > > Let me show you a specific scenario: Assuming that there is an article collection, there is a uniqueText field that specifies unique as true; Step 1: Perform a duplicate operation on the original article. At this time, the uniqueText field of the newly generated article becomes `-copy`. Step 2: **Go back to the original article again, pay attention! Here is to return to the original article** and continue to perform the duplicate operation. An error will be reported at this time. Because for uniqueText fields, `-copy` already exists.
> > 
> > 
> > Good point, you would use the Local API to check whether that slug already exists before returning the data. https://payloadcms.com/docs/local-api/overview#accessing-payload
> 
> One problem is that the parameters injected in the beforeDuplicate hook are only the data and locale fields, and the payload cannot be obtained for local api calls. In contrast, in the beforeValidate hook, I can get the payload (`req.payload.find(....)`) through the injected req parameter and make a local api call.
> 
> ![image](https://user-images.githubusercontent.com/16400882/196977332-b97f28b1-e4eb-4729-a929-baa367abf7e5.png) ![image](https://user-images.githubusercontent.com/16400882/196978249-3a785708-279a-4628-8d93-697383c2b562.png)

You can still import payload and use `payload.find` without directly accessing it through the request. Though it is interesting to why the request isn't passed here.

**Comment by zhounewz** [comment]
> > > > > > @jmikrut How to determine if `${data.uniqueText}-copy` is a duplicate, because it is possible
> > > > > > <img alt="image" width="1075" src="https://user-images.githubusercontent.com/16400882/196839358-0b90861a-02a3-4f0d-b237-a61b1b37361e.png">
> > > > > 
> > > > > 
> > > > > It's not possible because it will always append with `-copy` so it will become `-copy-copy`.
> > > > 
> > > > 
> > > > Let me show you a specific scenario: Assuming that there is an article collection, there is a uniqueText field that specifies unique as true; Step 1: Perform a duplicate operation on the original article. At this time, the uniqueText field of the newly generated article becomes `-copy`. Step 2: **Go back to the original article again, pay attention! Here is to return to the original article** and continue to perform the duplicate operation. An error will be reported at this time. Because for uniqueText fields, `-copy` already exists.
> > > 
> > > 
> > > Good point, you would use the Local API to check whether that slug already exists before returning the data. https://payloadcms.com/docs/local-api/overview#accessing-payload
> > 
> > 
> > One problem is that the parameters injected in the beforeDuplicate hook are only the data and locale fields, and the payload cannot be obtained for local api calls. In contrast, in the beforeValidate hook, I can get the payload (`req.payload.find(....)`) through the injected req parameter and make a local api call.
> > ![image](https://user-images.githubusercontent.com/16400882/196977332-b97f28b1-e4eb-4729-a929-baa367abf7e5.png) ![image](https://user-images.githubusercontent.com/16400882/196978249-3a785708-279a-4628-8d93-697383c2b562.png)
> 
> You can still import payload and use `payload.find` without directly accessing it through the request. Though it is interesting to why the request isn't passed here.

Hmm I didn't do it successfully, the following is my test code:

```
import payload from "payload";
import { BeforeDuplicate } from "payload/types";

export const beforeDuplicate: BeforeDuplicate = ({ data }) => {
  console.log('log payload:', payload, payload.find); // log payload: () => { } undefined
  return {
    ...data,
  };
};
```

**Comment by TomSmedley** [comment]
@zhounewz did you manage to figure this out?
@jmikrut does `beforeDuplicate` need to pass the request for this instance?

**Comment by thgh** [comment]
I tackled this today:
```ts
  hooks: {
    beforeOperation: [autoincrement('nummer')]
  }
```


```ts
function autoincrement(field: string): CollectionBeforeOperationHook {
  return async ({
    operation,
    args,
  }: {
    operation: string
    args?: { collection: any; data: Aanvraag }
  }) => {
    if (operation !== 'create') return args
    const slug = args.collection?.config?.slug
    const highest = await payload.find({
      collection: slug,
      limit: 1,
      sort: '-' + field,
    })
    const next = (parseInt(highest.docs[0]?.[field]) || 0) + 1
    return { ...args, data: { ...args.data, [field]: next } }
  }
}

```
